### PR TITLE
fix the localdisk.partitionInfo.path not display correctly

### LINKED ISSUE #1198: partitionInfo.Path doesn't have a correct value
The output of `kubectl get ld -oyaml` about the partitionInfo not show properly. The `partitionInfo.path` doesn't have a value.
```
  apiVersion: hwameistor.io/v1alpha1
  kind: LocalDisk
  metadata:
    creationTimestamp: "2023-10-06T13:17:38Z"
    generation: 2
    name: localdisk-18cf9ca870abe5cb1c1f2fca6105cb8b
    resourceVersion: "30044"
    uid: 3da242cf-6575-4021-98e2-09c9b35d15ba
  spec:
    capacity: 16106127360
    devLinks:
    - /dev/disk/by-id/ata-VMware_Virtual_IDE_Hard_Drive_00000000000000000001
    - /dev/disk/by-id/wwn-0x5000c298825951d9
    - /dev/disk/by-path/pci-0000:00:07.1-ata-1.0
    devicePath: /dev/sdb
    diskAttributes:
      devType: disk
      modelName: VMware_Virtual_IDE_Hard_Drive
      protocol: ata
      serialNumber: VMware_Virtual_IDE_Hard_Drive_00000000000000000001
      type: HDD
    major: "8"
    minor: "16"
    nodeName: kubemaster
    owner: system
    partitionInfo:
    - filesystem:
        fsType: LVM2_member
      hasFileSystem: true
      path: "" # here is an empty value
    partitioned: true
    raidInfo: {}
    reserved: true
    smartInfo:
      overallHealth: Failed
    state: Active
    uuid: 18cf9ca870abe5cb1c1f2fca6105cb8b
  status:
    claimState: Bound
```
The reason why this problem occur is the `partitionInfo.Path` not to be assign a proper value when it was appended to the list. But `manager.PartitionInfo` doesn't have a field liked `Path`, so the `Path` need to append and  keep the partition path value in order to the `localdisk.PartitionInfo` can have a proper display.
And  `Device` struct in udev directory have a field named `PartName` not to be initialized a value when parsing the result from the `udevadm info` command. Because the result of  `udevadm info` doesn't have a field like `PartName`, it must be assigned by programmer.

### PR DESCRIPTION
**Pull request by hikariwo**
#### What this PR does / why we need it:
fix #1198
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/local-disk-manager/udev/device_test.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package udev
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestDevice_ParseDeviceInfo_UnPartition(t *testing.T) {
```
Comment (SSmallMonster) [inline pkg/local-disk-manager/udev/device_test.go:8]: these 2 test cases may be merged into one TestFunc, that may be better.

File: pkg/local-disk-manager/disk/manager/partition.go [hunk 2]
```diff
@@ -29,4 +29,7 @@ type PartitionInfo struct {
 
 	// Filesystem
 	Filesystem string
+
+	// Path
+	Path string
```
Comment (SSmallMonster) [inline pkg/local-disk-manager/disk/manager/partition.go:34]: does this path represent the partition-path like `/dev/sda1`,`/dev/sda2`? 

i'm wondering if it's a mount-path, but I think the partition-path should be more meaningful.

**Review by SSmallMonster — COMMENTED** [review]
overall lgtm, just some nits

File: pkg/local-disk-manager/disk/manager/partition.go (hunk 2 above)
Comment (hikariwo) [reply pkg/local-disk-manager/disk/manager/partition.go:34]: Yes,its format turly like `/dev/sda1`. It comes from the field `DEVNAME`  of `udevadm info` command.

File: pkg/local-disk-manager/udev/device_test.go (hunk 1 above)
Comment (hikariwo) [reply pkg/local-disk-manager/udev/device_test.go:8]: ok, I try to modify it

File: pkg/local-disk-manager/udev/device_test.go [hunk 3]
```diff
@@ -5,23 +5,26 @@ import (
...
...sdb/sdb1\nE: DEVTYPE=partition\nE: ID_ATA=1\nE: ID_ATA_FEATURE_SET_APM=1\nE: ID_ATA_FEATURE_SET_APM_ENABLED=0\nE: ID_ATA_FEATURE_SET_PM=1\nE: ID_ATA_FEATURE_SET_PM_ENABLED=1\nE: ID_BUS=ata\nE: ID_FS_TYPE=LVM2_member\nE: ID_FS_USAGE=raid\nE: ID_FS_UUID=QKLBih-32Tt-0JP7-uQKr-GzjD-cx0C-I8Vnno\nE: ID_FS_UUID_ENC=QKLBih-32Tt-0JP7-uQKr-GzjD-cx0C-I8Vnno\nE: ID_FS_VERSION=LVM2 001\nE: ID_MODEL=VMware_Virtual_IDE_Hard_Drive\nE: ID_MODEL_ENC=VMware\\x20Virtual\\x20IDE\\x20Hard\\x20Drive\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\nE: ID_PART_ENTRY_DISK=8:16\nE: ID_PART_ENTRY_NUMBER=1\nE: ID_PART_ENTRY_OFFSET=2048\nE: ID_PART_ENTRY_SCHEME=dos\nE: ID_PART_ENTRY_SIZE=31455232\nE: ID_PART_ENTRY_TYPE=0x83\nE: ID_PART_TABLE_TYPE=dos\nE: ID_PATH=pci-0000:00:07.1-ata-1.0\nE: ID_PATH_TAG=pci-0000_00_07_1-ata-1_0\nE: ID_REVISION=00000001\nE: ID_SERIAL=VMware_Virtual_IDE_Hard_Drive_00000000000000000001\nE: ID_SERIAL_SHORT=00000000000000000001\nE: ID_TYPE=disk\nE: ID_WWN=0x5000c298825951d9\nE: ID_WWN_WITH_EXTENSION=0x5000c298825951d9\nE: MAJOR=8\nE: MINOR=17\nE: SUBSYSTEM=block\nE: SYSTEMD_ALIAS=/dev/block/8:17\nE: SYSTEMD_READY=1\nE: SYSTEMD_WANTS=lvm2-pvscan@8:17.service\nE: TAGS=:systemd:"
```
Comment (SSmallMonster) [inline pkg/local-disk-manager/udev/device_test.go:10]: you can use a struct to represent all test cases like https://github.com/hwameistor/hwameistor/blob/997ca1bb0d9c546c8a0cf34caecb97f6a8936dd1/pkg/local-disk-manager/handler/localdiskclaim/localdiskclaim_test.go#L49

File: pkg/local-disk-manager/udev/device_test.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+			ExpectResult: "sdb1",
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.Description, func(t *testing.T) {
+			info := parseUdevInfo(testCase.UdevInfo)
+			d := &Device{}
+			d.ParseDiskAttribute(info)
+			fmt.Printf("PartName: %v\n", d.PartName)
```
Comment (SSmallMonster) [inline pkg/local-disk-manager/udev/device_test.go:31]: this print log is unnecessry here

Comment (hikariwo) [reply pkg/local-disk-manager/udev/device_test.go:31]: ok,I remove it!

**Comment by SSmallMonster** [comment]
@hikariwo lgtm, can you rebase the commit and repush

**Comment by hikariwo** [comment]
I try to clean it!

> @hikariwo lgtm, can you rebase the commit and repush

**Comment by hikariwo** [comment]
@SSmallMonster done, commit-tree has been cleaned!

**Pull request merged**